pkg/backend/kube/backend.go

```diff
@@ -0,0 +1,225 @@
+package kube
+
+import (
+	"fmt"
+	"io/fs"
+	"net/http"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/Autodesk/shore/pkg/shore_testing"
+	jsoniter "github.com/json-iterator/go"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+)
+
+// SpinClient represents a client for managing pipelines.
+type SpinClient struct {
+	log logrus.FieldLogger
+}
+
+// NewClient - Create a new default spinnaker client
+func NewClient(logger logrus.FieldLogger) *SpinClient {
+	return &SpinClient{log: logger}
+}
+
+// SavePipeline - Saves the pipeline as a YAML file in the specified folder.
+func (s *SpinClient) SavePipeline(pipelineJSON string) (*http.Response, error) {
+	// Unmarshal the pipeline JSON to extract metadata
+
+	currentDir, err := os.Getwd()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get current directory: %w", err)
+	}
+
+	generateFolderPath := filepath.Join(currentDir, "generated")
+	if err := os.MkdirAll(generateFolderPath, os.ModePerm); err != nil {
+		return nil, fmt.Errorf("failed to create folder: %w", err)
+	}
+
+	var pipeline map[string]interface{}
+	if err := jsoniter.Unmarshal([]byte(pipelineJSON), &pipeline); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal pipeline JSON: %w", err)
+	}
+
+	// Iterate over the top-level keys and save each as a separate YAML file
+	for key, value := range pipeline {
+		// Convert the value to YAML
+		pipelineYAML, err := yaml.Marshal(value)
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal YAML for key %s: %w", key, err)
+		}
+
+		// Create the file path
+		filePath := filepath.Join(generateFolderPath, fmt.Sprintf("%s.yaml", key))
+
+		// Write the YAML to the file
+		if err := os.WriteFile(filePath, pipelineYAML, 0644); err != nil {
+			return nil, fmt.Errorf("failed to write YAML file for key %s: %w", key, err)
+		}
+
+		s.log.Infof("Saved YAML for key %s to %s", key, filePath)
+	}
+
+	return nil, nil
+}
+
+func (s *SpinClient) ExecutePipeline(argsJSON string, stringify bool) (string, *http.Response, error) {
+
+	// Check if a valid Kubernetes context is set
+	cmd := exec.Command("kubectl", "config", "current-context")
```

Comment:
Instead of relying on running a CLI command - let's use the native k8s client to do the apply - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.19.0/pkg/client#example-Client-Create